presto-main/src/test/java/io/prestosql/operator/scalar/AbstractTestRegexpFunctions.java

```diff
@@ -261,4 +264,108 @@ public void testRegexpSplit()
         assertFunction("REGEXP_SPLIT(',,a,,,b,c,d,,', ',')", new ArrayType(createVarcharType(13)), ImmutableList.of("", "", "a", "", "", "b", "c", "d", "", ""));
         assertFunction("REGEXP_SPLIT(',,,', ',')", new ArrayType(createVarcharType(3)), ImmutableList.of("", "", "", ""));
     }
+
+    @Test
+    public void testRegexpCount()
+    {
+        assertRegexpCount("a.b:c;d", "[\\.:;]", 3);
+        assertRegexpCount("a.b:c;d", "\\.", 1);
+        assertRegexpCount("a.b:c;d", ":", 1);
+        assertRegexpCount("a,b,c", ",", 2);
+        assertRegexpCount("a1b2c3d", "\\d", 3);
+        assertRegexpCount("a1b2346c3d", "\\d+", 3);
+        assertRegexpCount("abcd", "x", 0);
+        assertRegexpCount("Hello world bye", "\\b[a-z]([a-z]*)", 2);
+        assertRegexpCount("rat cat\nbat dog", "ra(.)|blah(.)(.)", 1);
+        assertRegexpCount("Baby X", "by ([A-Z].*)\\b[a-z]", 0);
+        assertRegexpCount("rat cat bat dog", ".at", 3);
+
+        assertInvalidFunction("REGEXP_EXTRACT('Hello world bye', '\\b[a-z]([a-z]*)', -1)", INVALID_FUNCTION_ARGUMENT);
+        assertInvalidFunction("REGEXP_EXTRACT('Hello world bye', '\\b[a-z]([a-z]*)', 2)", INVALID_FUNCTION_ARGUMENT);
+
+        assertRegexpCount("", "x", 0);
+
+        //Non-unicode string
+        assertRegexpCount("君子矜而不争，党而不群", "不", 2);
+
+        // empty pattern
+        assertRegexpCount("abcd", "", 5);
+
+        // sql in document
+        assertRegexpCount("1a 2b 14m", "\\s*[a-z]+\\s*", 3);
+    }
+
+    @Test
+    public void testRegexpPosition()
+    {
+        assertRegexpPosition("a.b:c;d", "[\\.:;]", 2);
+        assertRegexpPosition("a.b:c;d", "\\.", 2);
+        assertRegexpPosition("a.b:c;d", ":", 4);
+        assertRegexpPosition("a,b,c", ",", 2);
+        assertRegexpPosition("a1b2c3d", "\\d", 2);
+
+        // match from specified position
+        assertRegexpPosition("a,b,c", ",", 3, 4);
+        assertRegexpPosition("a1b2c3d", "\\d", 5, 6);
+
+        // match the n-th occurrence from from specified position
+        assertRegexpPosition("a1b2c3d4e", "\\d", 4, 2, 6);
+        assertRegexpPosition("a1b2c3d", "\\d", 4, 3, -1);
+
+        // empty pattern
+        assertRegexpPosition("a1b2c3d", "", 1);
+        assertRegexpPosition("a1b2c3d", "", 2, 2);
+        assertRegexpPosition("a1b2c3d", "", 2, 2, 3);
+        assertRegexpPosition("a1b2c3d", "", 2, 6, 7);
```

Comment:
I think this should be true also
```
assertRegexpPosition("a1b2c3d", "", 2, 7, 8)
```
Check against `regexp_replace`

Reply:
The result is `true`, I have added it into test cases.